src/app/services/crud.service.ts

```diff
@@ -2,63 +2,80 @@ import { Injectable } from '@angular/core';
 import { Observable, of } from 'rxjs';
 import {movies} from '../fake-data';
 import * as data from '../data-ter.json';
+import { AngularFirestore, AngularFirestoreDocument, AngularFirestoreCollection } from '@angular/fire/firestore';
+import { AngularFireObject, AngularFireDatabase } from '@angular/fire/database'
 
 
-export interface Movie {
-    id: number;
-    title: string;
-    languages: string;
-    producer: {
-      firstName: string,
-      lastName: string
-    };
- }
 
 @Injectable({
   providedIn: 'root'
 })
 export class CrudService {
 
-  public movies = data.movies;
-  public users = data.users;
-  constructor() { }
+  // public users = data.users;
+  movieRef: AngularFireObject<any>;
+  private movieDoc: AngularFirestoreDocument<any>;
+  moviesRef: AngularFirestoreCollection<any> = null;
+  constructor(private db: AngularFirestore, private rm: AngularFireDatabase) {}
 
-  addMovie(movie: Movie) {
-    this.movies.push({
-      id: movie.id,
-      title: movie.title,
-      languages: movie.languages,
-      producer: movie.producer
-    });
+  addMovie(movie: any) {
+    // this.movies.push({
+    //   id: movie.id,
+    //   title: movie.title,
+    //   languages: movie.languages,
+    //   producer: movie.producer
+    // });
   }
```

Comment:
There is too much dead code in this file, review it completely, delete comments that are not required or usefull, delete function that are not used anymore also.